olive/hf_utils.py

```diff
@@ -0,0 +1,98 @@
+from typing import Callable
+
+import torch
+
+
+def load_huggingface_model_from_task(task: str, name: str):
+    """Load huggingface model from task and name"""
+    from transformers import AutoModel
+    from transformers.pipelines import check_task
+
+    task_results = check_task(task)
+    assert isinstance(task_results, tuple)
+    if len(task_results) == 2:
+        targeted_task = task_results[0]
+    elif len(task_results) == 3:
+        targeted_task = task_results[1]
+    else:
+        raise ValueError("unsupported transfomers version")
+
+    model_class = {"pt": targeted_task["pt"]}
+    class_tuple = ()
+    class_tuple = class_tuple + model_class.get("pt", (AutoModel,))
+
+    model = None
+    for model_class in class_tuple:
+        try:
+            model = model_class.from_pretrained(name)
+            return model
+        except (OSError, ValueError):
+            continue
+
+    return model
+
+
+class ORTWhisperModel(torch.nn.Module):
+    """ORT implementation of whisper model"""
```

Comment:
I think moving this into a separate file under new "models" directory would make sense. We may have to offer custom nn.Modules in future also.